### PR TITLE
feat: Add dynamic credential provider image

### PR DESCRIPTION
**Pull request by jimmidyson**
**What problem does this PR solve?**:

**Which issue(s) does this PR fix?**:

* https://jira.d2iq.com/browse/D2IQ-96082

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

**Review by supershal — APPROVED** [review]

**Review by dkoshkin — APPROVED** [review]

File: ansible/group_vars/all/defaults.yaml
Comment (dkoshkin) [inline ansible/group_vars/all/defaults.yaml:61]: This looks good for building AMI and other cloud images when not air-gapped.
We do need to to also update the toml-merge image version and add [the new image in here](https://github.com/mesosphere/dkp-release/blob/a72a037fab7fd7bb9772bc352f5191e2539123fb/k8s-images-bundles/Makefile#L44).

Let's wait to merge this please until we have a new the RC cut.

Comment (dlipovetsky) [reply ansible/group_vars/all/defaults.yaml:61]: I think we can merge this once we have a `release-2.2` branch.

Comment (mkarroqe) [reply ansible/group_vars/all/defaults.yaml:61]: Images were updated and merged in mesosphere/dkp-release#66

**Comment by mkarroqe** [comment]
Waiting on mesosphere/konvoy-image-builder#748 to be merged to unblock this PR

**Comment by mkarroqe** [comment]
These changes were cherry-picked into #748- closing.

**Pull request closed without merging**
